src/hooks/use-create-user.tsx

```diff
@@ -0,0 +1,47 @@
+import { useMutation, gql, ApolloError } from "@apollo/client";
+import { useCallback } from "react";
+
+interface UserInput { 
+  name: string;
+  email: string;
+  phone: string;
+  birthDate: string;
+  password: string;
+  role: string;
+}
+
+interface CreateCallbacks {
```

Comment:
Aqui eu especificaria mais esse tipo:

```ts
interface CreateUserCallbacks {
  onCreateCompleted: (data?: UmTipoParaOUser) => void;
  onCreateError: (error?: ApolloError) => void;
}
```

Pq esse any pode não deixar muito claro para quem for utilizar esse hook o que ele envia nesse callback.